client/src/components/createcontact2.jsx

```diff
@@ -1,39 +1,117 @@
 
+function CreateContact2() {
+  // Use states
+  const [contacts, setContacts] = useState([]);
+  const [editContact, setEditContact] = useState(false);
+  const [currentContact, setCurrentContact] = useState({});
 
+  // Function to fetch contacts
+  const getRequest = () => {
+    fetch(`http://localhost:3000/api/contacts`)
+      .then((response) => response.json())
+      .then((contacts) => {
+        setContacts(contacts);
+        console.log('Contacts fetched...', contacts);
+      });
+  };
 
-function CreateContact2() {
+  // Function to add a new contact
+  const handlePOSTRequest = (data) => {
+    fetch("http://localhost:3000/api/contact", {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(data)
+    })
+      .then(response => response.json())
+      .then(data => {
+        console.log("inside the post line 30", data);
+        setContacts([...contacts, data]);
+      });
+  }
+
+  // Function to edit a contact
+  const modContact = (contact) => {
+    setEditContact(true);
+    setCurrentContact(contact);
+  };
 
-const relationships = ['friend', 'colleague', 'family', 'Business'];
+  // Function to update a contact
+  const updateContact = (id, updatedContact) => {
+    setContacts(contacts.map((contact) => (contact.id === id ? updatedContact : contact)));
+  };
 
-//USE STATES
-const [contact, setContact] = useState
+  const handleEditSubmit = async (updatedContact) => {
+    const response = await fetch(
+      `http://localhost:4000/api/contacts/${updatedContact.id}`,
+      {
+        method: 'PUT',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(updatedContact)
+      }
+    );
+    const result = await response.json();
 
-//CREATE FETCH REQUEST
-getRequest = async () => {
-    try {
-   const response = await fetch(`http://localhost:3000/api/contact`)
-        if(!response.ok) {
-            throw new Error(`Error! status: ${response.status}`);
-        }
-        const result = await response.json();
-        setContact(result);
-    }catch(err) {
-       console.error(err);
-    }
+    updateContact(result.id, updatedContact);
+    setCurrentContact(result);
 
-};
+    setEditContact(false);
+  };
 
-useEffect(() => {
+  // Use effect to fetch data when the component mounts
+  useEffect(() => {
     getRequest();
-},[]);
+  }, []);
+
+  return (
+    <div className="app-container">
+        <form onSubmit={handlePOSTRequest}>
+            <table>
+                <thead>
+                    <tr>
+                        <th>Name</th>
+                        <th>Email</th>
+                        <th>Phone Number</th>
+                        <th>Notes</th>
+                    </tr>
+               </thead>
+               <tbody>
+                {contacts.map((contact) => (
+                 <Fragment>
```

Comment:
curious as to what fragment is ?

Reply:
I need to delete that. It was apart of my previous attempt to use MUI into my project.